### PR TITLE
1556 Optimize the numpy computation for spatial axes

### LINKED ISSUE #1556: Random flip per channel seems redundant
A question from internal team:
https://github.com/Project-MONAI/MONAI/blob/bcdee8c062031a75e95cd777d7108d9db31970e2/monai/transforms/spatial/array.py#L285
 
random flip per channel seems redundant,  you can just flip once:
np.flip(img, self.spatial_axis+1)

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #1556 .

### Description
This PR optimized the numpy computation with a mapping utility for spatial axes.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/transforms/utils.py [hunk 1]
```diff
@@ -690,3 +691,42 @@ def extreme_points_to_image(
     points_image = (points_image - min_intensity) / (max_intensity - min_intensity)
     points_image = points_image * (rescale_max - rescale_min) + rescale_min
     return points_image
+
+
+def map_spatial_axes(
+    img: np.ndarray,
```
Comment (wyli) [inline monai/transforms/utils.py:697]: this could be `img_ndim: int`

Comment (Nic-Ma) [reply monai/transforms/utils.py]: Sure, update in the latest comment.
Thanks.

**Review by wyli — APPROVED** [review]

**Pull request merged**